core/common/src/main/java/alluxio/util/ShellUtils.java

```diff
@@ -193,6 +194,84 @@ private String run() throws ExitCodeException, IOException {
         process.destroy();
       }
     }
+
+    /**
+     * Runs a command and returns its stdout, stderr and exit code.
+     * No matter it succeeds or not.
+     *
+     * @return {@link CommandReturn} object representation of stdout, stderr and exit code
+     */
+    protected CommandReturn runTolerateFailure() throws IOException {
+      Process process = new ProcessBuilder(mCommand).start();
+      CommandReturn cr = null;
+
+      try (BufferedReader inReader =
+                   new BufferedReader(new InputStreamReader(process.getInputStream()));
+           BufferedReader errReader =
+                   new BufferedReader(new InputStreamReader(process.getErrorStream()))) {
+        // read the output of the command
+        StringBuilder stdout = new StringBuilder();
+        String outLine = inReader.readLine();
+        while (outLine != null) {
+          stdout.append(outLine);
+          stdout.append("\n");
+          outLine = inReader.readLine();
+        }
+
+        // read the stderr of the command
+        StringBuilder stderr = new StringBuilder();
+        String errLine = errReader.readLine();
+        while (errLine != null) {
+          stderr.append(errLine);
+          stderr.append("\n");
+          errLine = inReader.readLine();
+        }
+
+        // wait for the process to finish and check the exit code
+        int exitCode = process.waitFor();
+        if (exitCode != 0) {
+          // log error instead of throwing exception
+          LOG.warn(String.format("Failed to run command %s%nExit Code: %s%nStderr: %s",
+                  Arrays.toString(mCommand), exitCode, stderr.toString()));
+        }
+
+        cr = new CommandReturn(exitCode, stdout.toString(), stderr.toString());
+
+        // destroy the process
+        if (process != null) {
+          process.destroy();
+        }
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
+        throw new IOException(e);
+      } finally {
+        if (process != null) {
+          process.destroy();
+        }
```

Comment:
Is it okay to destroy the process and let the streams be managed by the try block?

In https://github.com/Alluxio/alluxio/blob/5c2bab41150b0d981238f21c3484e063e55f241a/core/common/src/main/java/alluxio/util/ShellUtils.java#L179 seems we tried to manually close the stream before destroying the process. Seems that's specific to certain Java 7 versions. Is this still the case now?